editor/src/clj/editor/app_view.clj

```diff
@@ -790,13 +791,40 @@
                                 :text "If the engine is already running, shut down the process manually and retry"}]}
            :content (.getMessage e)})))))
 
+(defn- owning-resource [basis node-id]
+  (when-some [owning-resource-node-id (core/scope-of-type basis node-id resource-node/ResourceNode)]
+             [owning-resource (resource-node/resource basis owning-resource-node-id)]
+    owning-resource))
+
+(defn- get-cycle-detected-help-message
+  [node-type-name node-id]
+  (let [basis (g/now)
+        proj-path (some-> (owning-resource basis node-id) resource/proj-path)
+        resource-path (some? proj-path) proj-path "'unknown'"]
+    (case node-type-name
+      "editor.collection/CollectionNode" (format "This is caused by a two or more collections referenced in a loop from either collection proxies or collection factories. One of the involved resources is: %s." resource-path)
+      "editor.game-object/GameObjectNode" (format "This is caused by two or more game objects referenced in a loop from game object factories. One of the involved resources is: %s." resource-path)
+      "editor.code.script/ScriptNode" (format "This is caused by two or more Lua modules required in a loop. One of the involved resources is: %s." resource-path)
+      (format "This is caused by two or more resources referenced in a loop. One of the involved resources is: %s." resource-path))))
+
```

Comment:
We can resolve the node type keyword from the `node-id` and add some whitespace here to make it more readable. This also lets us remove the `node-type-name` argument.

```suggestion
    (case (g/node-type-kw basis node-id)
      :editor.collection/CollectionNode
      (format "This is caused by a two or more collections referenced in a loop from either collection proxies or collection factories. One of the involved resources is: %s." resource-path)

      :editor.game-object/GameObjectNode
      (format "This is caused by two or more game objects referenced in a loop from game object factories. One of the involved resources is: %s." resource-path)

      :editor.code.script/ScriptNode
      (format "This is caused by two or more Lua modules required in a loop. One of the involved resources is: %s." resource-path)

      (format "This is caused by two or more resources referenced in a loop. One of the involved resources is: %s." resource-path))))
```